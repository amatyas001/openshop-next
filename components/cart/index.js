export * from './CartContent/CartContent';
export * from './CartAmount/CartAmount';
export * from './CartButton/CartButton';
export * from './CartItem/CartItem';
export * from './CartPanel/CartPanel';
