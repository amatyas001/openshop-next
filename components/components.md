Elements are composed by [Chakra-UI](https://chakra-ui.com) components and preserve the library features so [Style Props](https://chakra-ui.com/style-props) can be passed to them and accessibility remains optimized.

> _Style Props are a way to alter the style of a component by simply passing props to it. It helps to save time by providing helpful shorthand ways to style components._

For more information visit the [Chakra UI](https://chakra-ui.com) documentation.

Components are divided into two main category: _wrappers_ and _single responsibility_ elements. Wrapper components does not contain business logic, however there are special cases like _handling lifted states to keep sensitive data locally_ and not to populate them to the Redux store which is persisted on the machine.

**State Dependencies** and **Wrapped Components** are documented in each description to give brief information about the context of posible interactions of the actual component. _Examples_ prepared with proper state and may render extra controls to test different cases with ease.
