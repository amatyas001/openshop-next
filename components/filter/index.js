export * from './FilterButton/FilterButton';
export * from './FilterColor/FilterColor';
export * from './FilterPrice/FilterPrice';
export * from './FilterName/FilterName';
export * from './FilterPanel/FilterPanel';
export * from './FilterReset/FilterReset';
