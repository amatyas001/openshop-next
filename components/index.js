export * from './cart';
export * from './filter';
export * from './product';
export * from './layout';
export * from './payment';
export * from './shared';
