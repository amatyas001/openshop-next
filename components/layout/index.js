export * from './Content/Content';
export * from './Footer/Footer';
export * from './ProductsTopBar/ProductsTopBar';
export * from './Navbar/Navbar';
export * from './ScrollTop/ScrollTop';
