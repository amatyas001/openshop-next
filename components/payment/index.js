export * from './PaymentCancelled/PaymentCancelled';
export * from './PaymentConfirm/PaymentConfirm';
export * from './PaymentConfirmCard/PaymentConfirmCard';
export * from './PaymentConfirmControls/PaymentConfirmControls';
export * from './PaymentConfirmDetails/PaymentConfirmDetails';
export * from './PaymentError/PaymentError';
export * from './PaymentForm/PaymentForm';
export * from './PaymentFormControls/PaymentFormControls';
export * from './PaymentProgress/PaymentProgress';
export * from './PaymentReview/PaymentReview';
export * from './PaymentSuccess/PaymentSuccess';
