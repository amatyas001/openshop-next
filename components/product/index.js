export * from './ProductCard/ProductCard';
export * from './ProductControls/ProductControls';
export * from './ProductControlsColor/ProductControlsColor';
export * from './ProductControlsAmount/ProductControlsAmount';
export * from './ProductControlsButton/ProductControlsButton';
export * from './ProductControlsSize/ProductControlsSize';
