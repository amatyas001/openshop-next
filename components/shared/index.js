export * from './Button/Button';
export * from './ButtonIcon/ButtonIcon';
export * from './Field/Field';
export * from './Image/Image';
export * from './Panel/Panel';
export * from './Spinner/Spinner';
