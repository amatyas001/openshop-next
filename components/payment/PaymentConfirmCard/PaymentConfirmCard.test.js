import { create, act } from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { PaymentConfirmCard } from '@app/components';
import { INITIAL_STATE } from '@app/config';

const mockStore = configureStore([]);

const mock_load = jest.fn();
const mock_set = jest.fn();

const _state = {
  cart: [
    {
      id: 'mock_id',
      name: 'mock_name',
      desc: 'mock_desc',
      img: 'mock_img',
      price: 10,
      rating: 5,
    },
  ],
  payment: {
    status: 'confirm',
    details: {
      name: 'mock_name',
      email: 'mock_email',
      phone: 'mock_phone',
      address: 'mock_address',
    },
    intent: {
      id: 'mock_id',
      secret: 'mock_secret',
    },
    token: 'token',
  },
};

describe('<PaymentConfirmCard/>', () => {
  let tree, store;

  beforeAll(() => {
    store = mockStore(_state);

    act(() => {
      tree = create(
        <Provider store={store}>
          <PaymentConfirmCard loadHandler={mock_load} setHandler={mock_set} />
        </Provider>
      );
    });
  });

  it('should render with required props', () => {
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should lift handlers to parent', () => {
    expect(mock_set).toHaveBeenCalledTimes(1);
  });

  it('should set complete state', () => {
    act(() => {
      tree.root
        .findByProps({
          'data-testid': 'confirm-card-element',
        })
        .props.onChange({ complete: true });
    });

    expect(mock_set).toHaveBeenCalledTimes(2);
  });

  it('should display card error', () => {
    const mock_error = 'mock_error';

    act(() => {
      tree.root
        .findByProps({
          'data-testid': 'confirm-card-element',
        })
        .props.onChange({ error: { message: mock_error } });
    });

    expect(
      tree.root.findByProps({
        'data-testid': 'confirm-card-error',
      }).props.children
    ).toContain(mock_error);
  });

  it('should render with initial state', () => {
    act(() => {
      tree = create(
        <Provider store={mockStore(INITIAL_STATE)}>
          <PaymentConfirmCard loadHandler={mock_load} setHandler={mock_set} />
        </Provider>
      );
    });

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
