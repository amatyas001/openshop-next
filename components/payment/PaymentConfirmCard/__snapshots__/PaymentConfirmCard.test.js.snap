// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PaymentConfirmCard/> should render with initial state 1`] = `
<div
  className="css-1mi3tt8"
>
  <label
    className="css-8qu6k5"
    htmlFor="StripeElement"
  >
    enter your card details to finish payment
  </label>
  <div
    className="css-16soxp"
  >
    <CardElement
      data-testid="confirm-card-element"
      onChange={[Function]}
      options={
        Object {
          "iconStyle": "solid",
          "style": Object {
            "base": Object {
              "fontFamily": "Khand, sans-serif",
              "fontSize": "18px",
              "fontSmoothing": "antialiased",
              "fontWeight": "bold",
            },
          },
        }
      }
    />
  </div>
  <p
    className="css-165naot"
  >
    Use any future date and CVC with 
    <strong>
      4242 4242 4242 4242
    </strong>
     
    number for testing
  </p>
</div>
`;

exports[`<PaymentConfirmCard/> should render with required props 1`] = `
<div
  className="css-1mi3tt8"
>
  <label
    className="css-8qu6k5"
    htmlFor="StripeElement"
  >
    enter your card details to finish payment
  </label>
  <div
    className="css-16soxp"
  >
    <CardElement
      data-testid="confirm-card-element"
      onChange={[Function]}
      options={
        Object {
          "iconStyle": "solid",
          "style": Object {
            "base": Object {
              "fontFamily": "Khand, sans-serif",
              "fontSize": "18px",
              "fontSmoothing": "antialiased",
              "fontWeight": "bold",
            },
          },
        }
      }
    />
  </div>
  <p
    className="css-165naot"
  >
    Use any future date and CVC with 
    <strong>
      4242 4242 4242 4242
    </strong>
     
    number for testing
  </p>
</div>
`;
